powertools-serialization/src/main/java/software/amazon/lambda/powertools/utilities/EventDeserializer.java

```diff
@@ -0,0 +1,220 @@
+                    .map(r -> decode(r.getData()))
+                    .collect(Collectors.toList()));
+        } else if (object instanceof KinesisAnalyticsStreamsInputPreprocessingEvent) {
+            KinesisAnalyticsStreamsInputPreprocessingEvent event = (KinesisAnalyticsStreamsInputPreprocessingEvent) object;
+            return new EventPart(event.getRecords().stream()
+                    .map(r -> decode(r.getData()))
+                    .collect(Collectors.toList()));
+        } else {
+            // does not really make sense to use this EventDeserializer when you already have a typed object
+            // just not to throw an exception
+            LOG.warn("Consider using your object directly instead of using EventDeserializer");
+            return new EventPart(object);
+        }
+    }
+
+    /**
+     * Meaningful part of a Lambda event.<br/>
+     * Use {@link #extractDataFrom(Object)} to retrieve an instance of this class.
+     */
+    public static class EventPart {
+        private Map<String, Object> contentMap;
+        private String content;
+        private List<String> contentList;
+        private Object contentObject;
+
+        private EventPart(List<String> contentList) {
+            this.contentList = contentList;
+        }
+
+        private EventPart(String content) {
+            this.content = content;
+        }
+
+        private EventPart(Map<String, Object> contentMap) {
+            this.contentMap = contentMap;
+        }
+
+        private EventPart(Object content) {
+            this.contentObject = content;
+        }
+
+        /**
+         * Deserialize this part of event from JSON to an object of type T
+         * @param clazz the target type for deserialization
+         * @param <T> type of object to return
+         * @return an Object of type T (deserialized from the content)
+         */
+        public <T> T as(Class<T> clazz) {
+            try {
+                if (content != null) {
+                    if (content.getClass().equals(clazz)) {
+                        // do not read json when returning String, just return the String
+                        return (T) content;
+                    }
+                    return JsonConfig.get().getObjectMapper().reader().readValue(content, clazz);
+                }
+                if (contentMap != null) {
+                    return JsonConfig.get().getObjectMapper().convertValue(contentMap, clazz);
+                }
+                if (contentObject != null) {
+                    return (T) contentObject;
+                }
+                if (contentList != null) {
+                    throw new EventDeserializationException("The content of this event is a list, consider using 'asListOf' instead");
+                }
+                // should not occur, except if the event is malformed (missing fields)
+                throw new IllegalStateException("Event content is null");
```

Comment:
It will be great to put a bit meaningful message for the users as well in case this happens at all, Like what could be the reason as mentioned in comments?